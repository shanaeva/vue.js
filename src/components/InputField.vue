<template>
  <input
    type="text"
    class="input-field"
    :value="modelValue"
    :disabled="isDisable"
    @input="updateInput"
  >
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'InputField',
  props: {
    modelValue: { type: [String, Number], required: true },
    isDisable: {
      type: Boolean,
      require: false,
      default: false,
    },
  },
  emits: ['update:modelValue'],
  methods: {
    updateInput(event: {target: {value: string}}) {
      this.$emit('update:modelValue', event.target.value);
    },
  },
});
</script>

<style scoped>
  .input-field {
    padding: 16px;
    max-height: 60px;
    width: 100%;
    background-color: #424242;
    border-radius: 4px;
    border: none;
    color: #ffffff;
  }
  .input-field:focus-visible {
    border: none;
    outline: 1px solid #232323;
  }
</style>
