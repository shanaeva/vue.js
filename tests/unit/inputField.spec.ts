import { shallowMount } from '@vue/test-utils';
import InputField from '@/components/InputField.vue';

describe('InputField', () => {
  it('render', () => {
    const wrapper = shallowMount(InputField, { props: { isDisable: false, modelValue: 'text' } });
    expect(wrapper.props()).toStrictEqual({ isDisable: false, modelValue: 'text' });
  });
  it('render correctly', () => {
    const wrapper = shallowMount(InputField, { props: { isDisable: false, modelValue: 'text' } });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
  it('when enter text, input update', () => {
    const wrapper = shallowMount(InputField, { props: { isDisable: false, modelValue: 'text' } });
    wrapper.element.value = 'newValue';

    expect(wrapper.element.value).toBe('newValue');
  });
});
