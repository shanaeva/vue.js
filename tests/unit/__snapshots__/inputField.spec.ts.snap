// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField render correctly 1`] = `
<input
  class="input-field aqa-input-field"
  type="text"
/>
`;

exports[`InputField when enter text, input update 1`] = `
<input
  class="input-field aqa-input-field"
  type="text"
/>
`;
